sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -1471,14 +1475,14 @@ def pipeline_distribution_components(job_in_number, job_in_path):
             hello_world_component_mpi.distribution.process_count_per_instance = 2
 
         pipeline: PipelineJob = pipeline_distribution_components(10, job_input)
-        assert mpi_func._is_anonymous is False
```

Comment:
Why removed this?

Reply:
1. Given on-disk cache is enabled within a test, arm if of `mpi_func` will be from cache directly, while `mpi_func._is_anonymous` is set to `True` during `components.create_or_update`
2. `_is_anonymous` is a private implementation, so I change it to check public interface instead (whether it's a created anonymous component in the created job)

Reply:
Mixed the commented line. This assertion can be kept until we change related implementation. Rolled back this change.